integrationtest/osgi/karaf-it/src/test/java/org/hibernate/search/test/integration/osgi/HibernateSearchWithKarafIT.java

```diff
@@ -138,6 +148,24 @@ public void tearDown() {
 		bundleContext.ungetService( serviceReference );
 	}
 
+	/**
+	 * Starts all bundles to make sure there are no "uses constraint violations" caused by packages exported from the
+	 * bundles contributed by the HSEARCH Karaf feature.
+	 */
+	@Test
+	@TestForIssue(jiraKey="HSEARCH-1774")
+	public void testStartingAllBundles() throws Exception {
+		for ( Bundle bundle : bundleContext.getBundles() ) {
+			if ( !isFragmentBundle( bundle ) ) {
+				bundle.start();
+			}
+		}
+	}
+
+	private boolean isFragmentBundle(Bundle bundle) {
```

Comment:
> what's a fragment bundle?

It's a bundle which does not live on its own but attaches to another bundle
(the "fragment host") and makes its contents available to that host (see
http://wiki.osgi.org/wiki/Fragment). So it's a way to add contents to
another bundle. I'm excluding them here as they cannot be started on their
own.

Fragments may actually be useful for us if e.g. engine is a fragment host
and other HS modules are fragments of that. That way they can use e.g.
internal stuff from engine. I didn't go for this with the ISPN module
though as it's subject to extraction from the code base anyways.